cmd/thanos/bucket.go

```diff
@@ -377,6 +381,46 @@ func registerBucketWeb(m map[string]setupFunc, root *kingpin.CmdClause, name str
 	}
 }
 
+// Provide a list of resolution, can not use Enum directly, since string does not implement int64 function.
+func listResLevel() []string {
+	return []string{
+		strconv.FormatInt(downsample.ResLevel0, 10),
+		strconv.FormatInt(downsample.ResLevel1, 10),
+		strconv.FormatInt(downsample.ResLevel2, 10)}
+}
+
+func registerBucketReplicate(m map[string]setupFunc, root *kingpin.CmdClause, name string, objStoreConfig *extflag.PathOrContent) {
+	cmd := root.Command("replicate", fmt.Sprintf("Replicate data from one object storage to another. NOTE: Currently it works only with Thanos blocks (%v has to have Thanos metadata).", block.MetaFilename))
+	httpMetricsBindAddr, _ := regHTTPFlags(cmd)
+	toObjStoreConfig := regCommonObjStoreFlags(cmd, "-to", false, "The object storage which replicate data to.")
+	resolution := cmd.Flag("resolution", "Only blocks with this resolution will be replicated.").Default(strconv.FormatInt(downsample.ResLevel0, 10)).HintAction(listResLevel).Int64()
```

Comment:
hm, I think we might need actually `Int64s` for it as we might want to replicate more than one resolution. Let's add `TODO` for now.

Also `resolution-level` ?

```suggestion
        // TODO(bwplotka): Allow to replicate many compaction levels.
	resolution := cmd.Flag("resolution-level", "Only blocks with this resolution will be replicated.").Default(strconv.FormatInt(downsample.ResLevel0, 10)).HintAction(listResLevel).Int64()
```

Reply:
Not resolved

Reply:
In fact it does.

Reply:
How? I proposed `resolution-level` (: I might be wrong, but you should comment that I am wrong, because of X not ignore it. But I am pretty sure you just did not notice (:

Reply:
I thought that you mean add more level support for both resolution and compaction, so i added two TODOs here.

Reply:
Kinpin supports `Int` but not `Int64`, so still need to use `String` here.